File: UtilsLibrary/src/main/java/com/backpacker/UtilsLibrary/java/DbCommand.java
package com.backpacker.UtilsLibrary.java;

import android.os.Handler;
import android.os.Looper;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

/**
 * @version V 1.0 xxxxxxxx
 * @Title: kotlin_androidone
 * @Package com.backpacker.UtilsLibrary.java
 * @Description: 数局库异步线程
 * @author: L-BackPacker
 * @date: 2019/5/13 22:07
 * @verdescript 版本号 修改时间  修改人 修改的概要说明
 * @Copyright: 2019
 */
public abstract class DbCommand<T> {
    private final ExecutorService sDbEngsin= Executors.newCachedThreadPool();
    private final Handler mUiHandle=new Handler(Looper.getMainLooper());
    public final  void execute(){
        sDbEngsin.execute(new Runnable() {
            @Override
            public void run() {
                try {
                  doInbackground();
                }catch (Exception e){
                    e.printStackTrace();
                }
            }
        });
    }
    private void postReuslt(final T t){
        mUiHandle.post(new Runnable() {
            @Override
            public void run() {
                onPostExecture(t);
            }
        });
    }

    protected  abstract  void onPostExecture(T reuslt);
    protected abstract  T doInbackground();
}

File: UtilsLibrary/src/main/java/com/backpacker/UtilsLibrary/kotlin/DbCommand.kt
package com.backpacker.UtilsLibrary.kotlin

import android.os.Handler
import android.os.Looper
import android.os.Message
import java.util.concurrent.Executors

/**
 * @Title:  MyApplication
 * @Package com.example.myapplication
 * @Description:    $todo$
 * @author: L-BackPacker
 * @date:   2019/5/13 21:47
 * @version V 1.0 xxxxxxxx
 * @verdescript  版本号 修改时间  修改人 修改的概要说明
 * @Copyright: 2019
 */
abstract class DbCommand<T> {
    private val sDbMangerEngine = Executors.newCachedThreadPool()
    private val mUiHandler = Handler(Looper.getMainLooper())
    fun execture() {
        sDbMangerEngine.execute {
            try {
                doBackground()
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
    }

    fun postResult(t: T) {
        mUiHandler.post {
            onPostExecture(t)
        }
    }

    abstract fun doBackground(): T;
    abstract fun onPostExecture(result: T);
}